### DISCUSSION TITLE
cant manage to add static files/folders (or misunderstanding functionality)

### DISCUSSION THREAD

**Discussion by Liquidmasl** [zauberzeug/nicegui#1978] [Q&A] [answered]
### Description

hey guys and gals, i am at a loss here. can someone shine some knowledge at this issue?

I try to add a static folder to my site but it does not seam to appear [in the browser dev view "soureces"]

```py
abs_icon_path = os.path.join(os.path.dirname(__file__), 'qapture_dark_mode.ico')
app.add_static_file(url_path='/qapture_dark_mode.ico', local_file=abs_icon_path)

abs_static_path = os.path.join(os.path.dirname(__file__), 'static')
app.add_static_files(url_path='/static', local_directory=abs_static_path)

abs_style_path = os.path.join(os.path.dirname(__file__), 'styles.css')
app.add_static_file(url_path='/styles.css', local_file=abs_style_path)
```

here is what i tried. 
with those lines, no file or folder gets added to "sources" in the dev view

if I add 

```py
ui.add_head_html('''<link rel="stylesheet" type="text/css" href="styles.css">''')
```

suddenly the .css appear in sources. 
So that makes me think, is somehow the browser only loading what is referenced? Some optimization?
and thats why it did not appear before in sources?
was it there all along?

adding
```py
ui.html('''
<img src = "static/icons/svg_dark_mode.svg" alt="My Happy SVG"/>
''')
```
nothing more happens. i just get a 404 in the console cause the file wasnt found

same thing for 
```py
ui.html('''
<img src = "svg_dark_mode.svg" alt="My Happy SVG"/>
''')
```

Now I dont know what is failing. is it me not understanding how the browser works? or nicegui?
how can i check the files on the server directly?

I sit way to long at this. 
how can i just display this svg :')

**Comment by falkoschindler** [comment, marked as answer]
Adding static files does not guarantee that the client will actually request them. So unless you add an element which needs these files, they won't appear in "sources".

But with `ui.html` it should work. Maybe your paths are mixed up somehow? Let's break it down:
```py
app.add_static_files(url_path='/static', local_directory=os.path.join(os.path.dirname(__file__), 'static'))
```
This should add a route "/static" which points to a folder "static" in the directory of the Python file.
```py
ui.html('<img src="static/icons/svg_dark_mode.svg" alt="My Happy SVG" />')
```
Now this should load the SVG from the file "static/icons/svg_dark_mode.svg", since "/static" serves the folder "static".

Do you have different pages? Maybe you need a leading slash "/" for the image source.

**Reply by Liquidmasl** [reply]
thank you for moving here, and sorry for posting as an issue in the first place. 

As you said, it was a stupid user error. After I read through your response and retried all steps i found the issue. it never was an svg. its an .ico. 

sorry, it works now!

**Reply by falkoschindler** [reply]
No worries! Glad you found the error. 😊
